docs/oppen.md

```diff
@@ -0,0 +1,82 @@
+# Derek C. Oppen's pretty pritning algorithm
+
+Derek C. Oppen's pretty printing algorithm is a method for formatting code to make it more readable and visually appealing. It arranges code with indentation and line breaks. Oppen's algorithm has influenced many tools for code formatting and remains an important technique in software development for improving code clarity.
+
+## The printing tokens
+
+### String
+
+...?
+
+### Begin
+
+...?
+
+### End
+
+...?
+
+### Break
+
+...?
+
+### LineBreak
+
+...?
+
+## How it works
+
+### The cornerstones
```

Comment:
no need for this heading, just remove this, and move the ones 4 levels deep, one level up.